### ISSUE TITLE
Fail to detect 'bsd-new' and 'bsd-original-uc'

### ISSUE THREAD

**Issue by chinyeungli** [nexB/scancode-toolkit#299]
```
/*
 * sniffex.c
 *
 * Sniffer example of TCP/IP packet capture using libpcap.
 * 
 * Version 0.1.1 (2005-07-05)
 * Copyright (c) 2005 The Tcpdump Group
 *
 * This software is intended to be used as a practical example and 
 * demonstration of the libpcap library; available at:
 * http://www.tcpdump.org/
 *
 ****************************************************************************
 *
 * This software is a modification of Tim Carstens' "sniffer.c"
 * demonstration source code, released as follows:
 * 
 * sniffer.c
 * Copyright (c) 2002 Tim Carstens
 * 2002-01-07
 * Demonstration of using libpcap
 * timcarst -at- yahoo -dot- com
 * 
 * "sniffer.c" is distributed under these terms:
 * 
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 * 1. Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 * 4. The name "Tim Carstens" may not be used to endorse or promote
 *    products derived from this software without prior written permission
 *
 * THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND
 * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
 * ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
 * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
 * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
 * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
 * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
 * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
 * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 * SUCH DAMAGE.
 * <end of "sniffer.c" terms>
 *
 * This software, "sniffex.c", is a derivative work of "sniffer.c" and is
 * covered by the following terms:
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 * 1. Because this is a derivative work, you must comply with the "sniffer.c"
 *    terms reproduced above.
 * 2. Redistributions of source code must retain the Tcpdump Group copyright
 *    notice at the top of this source file, this list of conditions and the
 *    following disclaimer.
 * 3. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 * 4. The names "tcpdump" or "libpcap" may not be used to endorse or promote
 *    products derived from this software without prior written permission.
 *
 * THERE IS ABSOLUTELY NO WARRANTY FOR THIS PROGRAM.
 * BECAUSE THE PROGRAM IS LICENSED FREE OF CHARGE, THERE IS NO WARRANTY
 * FOR THE PROGRAM, TO THE EXTENT PERMITTED BY APPLICABLE LAW.  EXCEPT WHEN
 * OTHERWISE STATED IN WRITING THE COPYRIGHT HOLDERS AND/OR OTHER PARTIES
 * PROVIDE THE PROGRAM "AS IS" WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED
 * OR IMPLIED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
 * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.  THE ENTIRE RISK AS
 * TO THE QUALITY AND PERFORMANCE OF THE PROGRAM IS WITH YOU.  SHOULD THE
 * PROGRAM PROVE DEFECTIVE, YOU ASSUME THE COST OF ALL NECESSARY SERVICING,
 * REPAIR OR CORRECTION.
 * 
 * IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
 * WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MAY MODIFY AND/OR
 * REDISTRIBUTE THE PROGRAM AS PERMITTED ABOVE, BE LIABLE TO YOU FOR DAMAGES,
 * INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES ARISING
 * OUT OF THE USE OR INABILITY TO USE THE PROGRAM (INCLUDING BUT NOT LIMITED
 * TO LOSS OF DATA OR DATA BEING RENDERED INACCURATE OR LOSSES SUSTAINED BY
 * YOU OR THIRD PARTIES OR A FAILURE OF THE PROGRAM TO OPERATE WITH ANY OTHER
 * PROGRAMS), EVEN IF SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE
 * POSSIBILITY OF SUCH DAMAGES.
 * <end of "sniffex.c" terms>
 * 
```
No license information is returned for the above text (http://www.tcpdump.org/sniffex.c)
```
  "results": [
    {
      "location": "C:/license_texts/multi-bsd.license",
      "copyrights": [
        {
          "statements": [
            "Copyright (c) 2005 The Tcpdump Group"
          ],
          "holders": [
            "Tcpdump Group"
          ],
          "authors": [],
          "start_line": 6,
          "end_line": 7
        },
        {
          "statements": [
            "Copyright (c) 2002 Tim Carstens 2002-01-07"
          ],
          "holders": [
            "Tim Carstens"
          ],
          "authors": [],
          "start_line": 18,
          "end_line": 22
        }
      ],
      "licenses": [],
      "packages": []
    }
  ]
```

**Comment by pombredanne** [comment]
@chinyeungli Thank you for the report. Are you using the develop branch? if so as of when? or which version?

**Comment by chinyeungli** [comment]
```
(scancode-toolkit) C:\...\scancode-toolkit>scancode --version
ScanCode version 1.6.3
```
I did the pull couple days ago.

**Comment by pombredanne** [comment]
Do you mind trying with the latest from develop?

**Comment by chinyeungli** [comment]
sure, I will try later

**Comment by chinyeungli** [comment]
```
{
      "location": "C:/Users/CYL/nexB/ARRIS/Projects/Current/WHS_Gateway-Player-ARFQ-286/inventory/license_texts/multi-bsd.license",
      "copyrights": [
        {
          "statements": [
            "Copyright (c) 2005 The Tcpdump Group"
          ],
          "holders": [
            "Tcpdump Group"
          ],
          "authors": [],
          "start_line": 6,
          "end_line": 7
        },
        {
          "statements": [
            "Copyright (c) 2002 Tim Carstens 2002-01-07"
          ],
          "holders": [
            "Tim Carstens"
          ],
          "authors": [],
          "start_line": 18,
          "end_line": 22
        }
      ],
      "licenses": [
        {
          "key": "bsd-simplified",
          "score": 88.04,
          "short_name": "BSD-Simplified",
          "category": "Attribution",
          "owner": "Regents of the University of California",
          "homepage_url": "http://www.opensource.org/licenses/BSD-2-Clause",
          "text_url": "http://opensource.org/licenses/bsd-license.php",
          "dejacode_url": "https://enterprise.dejacode.com/license_library/Demo/bsd-simplified/",
          "spdx_license_key": "BSD-2-Clause",
          "spdx_url": "http://spdx.org/licenses/BSD-2-Clause",
          "start_line": 27,
          "end_line": 47
        },
        {
          "key": "bsd-simplified",
          "score": 17.39,
          "short_name": "BSD-Simplified",
          "category": "Attribution",
          "owner": "Regents of the University of California",
          "homepage_url": "http://www.opensource.org/licenses/BSD-2-Clause",
          "text_url": "http://opensource.org/licenses/bsd-license.php",
          "dejacode_url": "https://enterprise.dejacode.com/license_library/Demo/bsd-simplified/",
          "spdx_license_key": "BSD-2-Clause",
          "spdx_url": "http://spdx.org/licenses/BSD-2-Clause",
          "start_line": 53,
          "end_line": 60
        },
        {
          "key": "bsd-simplified",
          "score": 24.46,
          "short_name": "BSD-Simplified",
          "category": "Attribution",
          "owner": "Regents of the University of California",
          "homepage_url": "http://www.opensource.org/licenses/BSD-2-Clause",
          "text_url": "http://opensource.org/licenses/bsd-license.php",
          "dejacode_url": "https://enterprise.dejacode.com/license_library/Demo/bsd-simplified/",
          "spdx_license_key": "BSD-2-Clause",
          "spdx_url": "http://spdx.org/licenses/BSD-2-Clause",
          "start_line": 61,
          "end_line": 81
        }
      ],
      "packages": []
    }
```
It is giving wrong information as the license is not under `bsd-simplified`

```
(scancode-toolkit) C:\Users\CYL\git\scancode-toolkit>scancode --version
ScanCode version 2.0.0.dev0
```

**Comment by pombredanne** [comment]
So this is a bsd-new with something else derived from a bsd-new. Latest detection gets both ass bsd-new which is not entirely right but not incorrect either.

I am creating a new license for the second notice which is rather unique.

**Comment by pombredanne** [comment]
This is fixed and detected this way:
```
 "files": [
    {
      "path": "tests/licensedcode/data/licenses/sniffex.c", 
      "licenses": [
        {
          "key": "bsd-new", 
          "score": 100.0, 
          "short_name": "BSD-Modified", 
          "category": "Attribution", 
          "owner": "Regents of the University of California", 
          "homepage_url": "http://www.opensource.org/licenses/BSD-3-Clause", 
          "text_url": "", 
          "dejacode_url": "https://enterprise.dejacode.com/urn/urn:dje:license:bsd-new", 
          "spdx_license_key": "BSD-3-Clause", 
          "spdx_url": "http://spdx.org/licenses/BSD-3-Clause", 
          "start_line": 24, 
          "end_line": 47, 
          "matched_rule": {
            "identifier": "bsd-new_104.RULE", 
            "license_choice": false, 
            "licenses": [
              "bsd-new"
            ]
          }
        }, 
        {
          "key": "bsd-new-derivative", 
          "score": 100.0, 
          "short_name": "BSD-Derivative", 
          "category": "Attribution", 
          "owner": "tcpdump", 
          "homepage_url": "http://www.tcpdump.org/sniffex.c", 
          "text_url": "", 
          "dejacode_url": "https://enterprise.dejacode.com/urn/urn:dje:license:bsd-new-derivative", 
          "spdx_license_key": "", 
          "spdx_url": "", 
          "start_line": 53, 
          "end_line": 86, 
          "matched_rule": {
            "identifier": "bsd-new-derivative.LICENSE", 
            "license_choice": false, 
            "licenses": [
              "bsd-new-derivative"
            ]
          }
        }
      ]
    }
  ]

```

**Comment by pombredanne** [comment]
Fixed in #334
